lib/react-markdown.js

```diff
@@ -141,14 +141,48 @@ ReactMarkdown.propTypes = {
     PropTypes.oneOfType([
       PropTypes.object,
       PropTypes.func,
-      PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.func]))
+      PropTypes.arrayOf(
+        PropTypes.oneOfType([
+          PropTypes.bool,
+          PropTypes.number,
+          PropTypes.string,
+          PropTypes.object,
+          PropTypes.func,
+          PropTypes.arrayOf(
+            PropTypes.oneOfType([
+              PropTypes.bool,
+              PropTypes.number,
+              PropTypes.string,
+              PropTypes.object,
+              PropTypes.func
+            ])
+          )
+        ])
+      )
     ])
   ),
   rehypePlugins: PropTypes.arrayOf(
     PropTypes.oneOfType([
       PropTypes.object,
       PropTypes.func,
-      PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.func]))
+      PropTypes.arrayOf(
+        PropTypes.oneOfType([
+          PropTypes.bool,
+          PropTypes.number,
```

Comment:
Not on our side, but it seemed worth having the completeness? There is nothing wrong with a plugin taking numbers or booleans as options?

Reply:
> I agree that any is a better fit. Isn't it what the typescript type declarations have, anyway? Is this the right type to be looking at?

Not quite. Walking through that:
> `<PluginParameters`

Means that TypeScript wants a specific type

> `extends any[]`

means that TypeScript means that a plugin can have from 0 to any number of settings

> = any[]>

means for plugins without typings, don't type check.

Practically that means TypeScript will match an exact type based off what the plugin passes, but can fallback for legacy plugins which are untyped.

---
> > […] I don't see any as being a good fit
> >
> > I'm […] cautious adding […] complexity to the prop types, […] the TypeScript types offer a more precise understanding of what is accepted.
>
> I feel most for the latter. In particular because TS types are better and this current prop-types code is overlong, what about using “any”s here?

I'd see more a balance of these.
Maybe using `any` for one of the inner-more types?
For example:

```typescript
  // Plugin options:
  remarkPlugins: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.func,
      PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.string,
          PropTypes.object,
          PropTypes.func,
          PropTypes.arrayOf(
            PropTypes.any
          )
        ])
      )
    ])
```

Reply:
That seems like a reasonable balance. I don't think we should strive too hard to try to do what a real type checker does. 

re: Pluggable's any, point taken. My point was that we allow pluggables to have any types (and then, yes, hold them to their contract)... so we should do the same with react's pseudo type system.

Reply:
That is acceptable to me, though there are some valid edge cases where that still results in errors (booleans particularly) (false-positives) and there are several things that are non-sensical that prop-types allows anyway (`[settings, somePlugin]` etc). I weigh the false-positives more heavily than attempting to get closer to runtime/TS, so I have a slight preference for doing away with the `PropTypes.oneOfType([PropTypes.string ...])` bit.

Reply:
Would re-adding boolean to the initial list cover the false positive concern?

```ts
  // Plugin options:
  remarkPlugins: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.object,
      PropTypes.func,
      PropTypes.arrayOf(
        PropTypes.oneOfType([
          PropTypes.string,
          PropTypes.boolean,
          PropTypes.object,
          PropTypes.func,
          PropTypes.arrayOf(
            PropTypes.any
          )
        ])
      )
    ])
```

Reply:
That is also acceptable to me. Boolean is most important, but a) I’ve also seen userland plugins accept other random stuff (which is slightly weird so it’s not very important), b) the more stuff is added, the closer it becomes to `any` anyway

Reply:
I pushed this update. I had to add a pragma around the use of `PropTypes.any`:

```
// type-coverage:ignore-next-line  
```

Reply:
well that's bizarre. the code formatter you use changed the case of the pragmas, thus breaking the pragmas.

Reply:
The ESLint rule turned on by `xo` does unfortunately not understand `type-coverage` comments.
https://github.com/xojs/eslint-config-xo/blob/0a302bdd73bfd49e956ec7f5ebf2c2247c62c310/index.js#L274-L284.
You could try and PR it to them, or you can turn it off here.